armi/settings/setting.py

```diff
@@ -48,20 +46,20 @@ class Setting:
     """
     A particular setting.
 
-    Setting objects hold all associated information of a setting in ARMI and should
-    typically be accessed through the Settings class methods rather than directly. The
-    exception being the SettingAdapter class designed for additional GUI related
-    functionality.
-
-    Setting subclasses can implement custom ``load`` and ``dump`` methods
-    that can enable serialization (to/from dicts) of custom objects. When
-    you set a setting's value, the value will be unserialized into
-    the custom object and when you call ``dump``, it will be serialized.
-    Just accessing the value will return the actual object in this case.
-
     .. impl:: The setting default is mandatory.
         :id: I_ARMI_SETTINGS_DEFAULTS
         :implements: R_ARMI_SETTINGS_DEFAULTS
+
+        Setting objects hold all associated information of a setting in ARMI
+        and should typically be accessed through the Settings class methods
+        rather than directly. The exception being the SettingAdapter class
+        designed for additional GUI related functionality.
+
+        Setting subclasses can implement custom ``load`` and ``dump`` methods
+        that can enable serialization (to/from dicts) of custom objects. When
+        you set a setting's value, the value will be unserialized into
+        the custom object and when you call ``dump``, it will be serialized.
+        Just accessing the value will return the actual object in this case.
```

Comment:
This is just about the setting object in general, so I think you can move it back to where it was. Or leave it, fine by me. 

Just add that a setting object requires a default as mandatory for initialization. I really don't think there's much else to say TBH! Maybe you could add an example of a setting definition with a default, but I don't know how necessary it is. Or point to a settings definition file?

Reply:
I'll add the default. The extra information is here by request.